src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```diff
@@ -1232,6 +1280,64 @@ because xbuild doesn't support framework reference assemblies.
 	</ResolveLibraryProjectImports>
 </Target>
 
+<Target Name="_CollectLibraryResourceDirectories">
+	<CollectNonEmptyDirectories Directories="@(LibraryResourceDirectories);@(_AdditonalAndroidResourceCachePaths->'%(Identity)\res')">
+		<Output TaskParameter="Output" ItemName="_LibraryResourceDirectories" />
+	</CollectNonEmptyDirectories>
+</Target>
+
+<Target Name="_CompileAndroidLibraryResources" DependsOnTargets="_CollectLibraryResourceDirectories"
+		Condition="'$(AndroidUseAapt2)' == 'True' And '$(DesignTimeBuild)' != 'True'"
+		Inputs="%(_LibraryResourceDirectories.FileFound)"
+		Outputs="%(_LibraryResourceDirectories.Identity)\..\compiled.stamp"
+	>
+	<ConvertResourcesCases
+		ResourceDirectories="@(_LibraryResourceDirectories)"
+		AcwMapFile="$(_AcwMapFile)"
+		AndroidConversionFlagFile="%(_LibraryResourceDirectories.Identity)\..\compiled.flata"
+		ContinueOnError="$(DesignTimeBuild)"
+	/>
+	<Aapt2Compile
```

Comment:
Given that this is being used in a "batching" context, should the `<Aapt2Compile/>` and `<Aapt2/>` task instead follow the example of `<Aapt/>`/`<Aot/>` and run the `aapt2 compile` commands *in parallel* within the task, instead of using MSBuild task batching?